tests/components/deako/test_light.py

```diff
@@ -0,0 +1,259 @@
+"""Tests for the light module."""
+
+from unittest.mock import MagicMock
+
+import pytest
+
+from homeassistant.components.deako.const import DOMAIN
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    DOMAIN as LIGHT_DOMAIN,
+    ColorMode,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.asyncio
+async def test_light_setup_no_devices(
+    hass: HomeAssistant,
+    pydeako_deako_mock: MagicMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test light platform setup with no devices returned."""
```

Comment:
This test is strange because this will never happen in real life because init has a check for it

Reply:
Yeah you're right. Was more of an isolated test, not necessarily practical